packages/base-types/src/models/version/index.ts

```diff
@@ -53,4 +53,7 @@ export interface Model<_PlatformData extends PlatformData, Command extends BaseC
   prototype?: Prototype<Command, Locale>;
   components?: FolderItem[];
   platformData: _PlatformData;
+
+  manualSave?: boolean;
+  autoSaveFromRestore?: boolean;
```

Comment:
i'm not a big fan of optional booleans, it can be easy to make mistakes with the difference between `undefined` and `false`. is it possible to make these required properties? if not, does it make sense to document what the behavior should be when they are undefined?

Reply:
The optional property was primarily done because this is how it is done in MongoDB currently. Unlike the example I posted on JIRA, the `manualSave` and `autoSaveFromRestore` is allowed to be `undefined`. In fact, most versions being generated have it as `undefined`. It's only those versions that are created explicitly by the user that define this property. 

I don't believe there was any migration done to explicitly set this property to a non-`undefined` value for all existing documents. 

Documentation might not work, because we don't have an agreed-upon single source of truth, so it is too easy for a developer to overlook the documented behaviour. But for your reference, the `manualSave` and `autoSaveFromRestore` can be assumed to be `false` if it is `undefined`. 

We would either need to:
1. Perform a migration to modify all of the existing MongoDB documents, which I'm hesitant to do, because this change is relating to Production Version Management which is already expanded in scope - OR -
2. Change `creator-api` to return `!!manualSave` and `!!autoSaveFromRestore` to make these guarantees on the code side. I could try to do this on Monday.

Reply:
in that case let's keep this as-is for now, to reduce scope creep. could you please make a cooldown ticket to look into implementing one of those fixes you suggested?

Reply:
Sure understood.